code/class/registry/registry.php

```diff
@@ -129,7 +136,54 @@ public function alias($class, $alias)
     }
 
     /**
-     * Get a list of all the identifier aliases
+     * Get a specific locator from the class namespace
+     *
+     * @return string|false The name of the locator or FALSE if no locator could be found
+     */
+    public function getLocator($class)
+    {
+        $result = false;
+
+        if(!false == $pos = strrpos($class, '\\'))
+        {
+            $namespace = substr($class, 0, $pos);
+
+            if(isset($this->_namespaces[$namespace])) {
+                $result = $this->_namespaces[$namespace];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Set a specific locator based on a class namespace
+     *
+     * @return  ClassRegistry
+     */
+    public function setLocator($class, $locator)
+    {
+        if(!false == $pos = strrpos($class, '\\'))
```

Comment:
Because the registry is mapping `Namespace` => `LocatorName`. Whereas the `ClassLoader` allows the registration of multiple locators, which may all share a common namespace. However, once that namespace is resolved to a locator once, only the first locator that returns a match for the specific class will be cached.